app/models/restaurant_wheel.rb

```diff
@@ -30,17 +40,22 @@ def filtered_restaurants
     @filtered_restaurants
   end
 
-  def ordered_restaurants
-    @ordered_restaurants ||= filtered_restaurants.sort_by(&:average_rating).reverse
+  def total_average_rating
+    @total_average_rating ||= filtered_restaurants.sum(&:average_rating).to_d
   end
 
-  def generate_wheel
-    raise 'Must have restuarants to create a wheel!' if ordered_restaurants.empty?
-
-    @wheel = []
-    while @wheel.count < num_stops
-      @wheel += ordered_restaurants
+  def cummulative_weighted_restaurants
+    total = 0
+    @cummulative_weighted_restaurants ||= filtered_restaurants.map do |restaurant|
+      total += restaurant.average_rating / total_average_rating
```

Comment:
this isn't actually weighting the averages, right? a 2.5 restaurant with 10 reviews would be given equal space to a 2.5 with none. although, now that i think about it, should we even weight them? and to what degree should a 5-star 10-review spot be given more space than a 5 star with only one review?